packages/simcore-sdk/src/simcore_sdk/node_ports_common/data_items_utils.py

```diff
@@ -1,14 +1,23 @@
 import tempfile
 import threading
 from pathlib import Path
+from typing import Optional
 
 from models_library.projects_nodes_io import SimcoreS3FileID
 
 
 def create_simcore_file_id(
-    file_path: Path, project_id: str, node_id: str
+    file_path: Path,
+    project_id: str,
+    node_id: str,
+    *,
+    file_base_path: Optional[Path] = None,
 ) -> SimcoreS3FileID:
-    return SimcoreS3FileID(f"{project_id}/{node_id}/{file_path.name}")
+    s3_file_name = file_path.name
+    if file_base_path:
+        s3_file_name = f"{file_base_path / file_path.name}"
+    clean_path = Path(f"{project_id}/{node_id}/{s3_file_name}")
+    return SimcoreS3FileID(f"{clean_path}")

```

Comment:
I see, this is the actual fix. 👍

Reply:
not exactly all of it... storage also needed a bit of love to accept this